pkg/client/k8s.go

```diff
@@ -743,3 +743,21 @@ func (k K8sClient) GetPods(labelSelector string, fieldSelector string, namespace
 	}
 	return pods, nil
 }
+
+/*
+	LOCAL VOLUMES SPECIFIC METHODS
+*/
+
+// GetLocalPvDeployment returns the deployments with a specific
+// openebs-component-name label key
+func (k K8sClient) GetLocalPvDeployment(name string) (*appsv1.Deployment, error) {
+	if pv, err := k.K8sCS.AppsV1().Deployments("").List(context.TODO(), metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("openebs.io/component-name=%s", name),
+	}); err == nil && len(pv.Items) == 1 {
+		return &pv.Items[0], nil
```

Comment:
@Abhinandan-Purkait Should the function just return `*appsv1.DeploymentList, error` instead, I think we did similar kind of enhancements.

Reply:
Yes, it can. At the point of calling we can extract the first element, as per this use case.

Reply:
Applied this :smiley: